futures/deployment/src/main/scala/io/quarkiverse/scala/scala3/futures/deployment/Scala3FutureResponseHandler.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkiverse.scala.scala3.deployment;
+
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.spi.ServerRestHandler;
+
+public class Scala3FutureResponseHandler implements ServerRestHandler {
```

Comment:
I think this is probably why you had to add the -deployment dependencies in your build files: this runtime class should be in a `runtime` module.

The `-deployment` module is only around at build time, anything that is used at runtime has to be in the runtime module.